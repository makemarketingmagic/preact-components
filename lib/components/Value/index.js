"use strict";

exports.__esModule = true;
exports.default = undefined;

var _preact = require("preact");

var _Value = require("./Value.less");

var _Value2 = _interopRequireDefault(_Value);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Value = function (_Component) {
    _inherits(Value, _Component);

    function Value() {
        _classCallCheck(this, Value);

        return _possibleConstructorReturn(this, _Component.apply(this, arguments));
    }

    Value.prototype.render = function render() {
        var children = this.props.children;

        return (0, _preact.h)(
            "div",
            { "class": _Value2.default.value },
            children
        );
    };

    return Value;
}(_preact.Component);

exports.default = Value;