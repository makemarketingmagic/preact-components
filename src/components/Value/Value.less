@import '../common/mixins.less';
@import '../common/variables.less';

.value {
    font-size: 24px;
    line-height: 32px;
    color: @text;
}