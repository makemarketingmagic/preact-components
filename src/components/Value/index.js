import { h, Component } from "preact";
import styled from 'styled-components';
import { colors } from "../common/scMixins";

const ValueEl = styled.div`
    font-size: 24px;
    line-height: 32px;
    color: ${colors.text};
`

export default class Value extends Component {
    render() {
        const { children } = this.props
        return (
            <ValueEl>
                {children}
            </ValueEl>
        )
    }
}