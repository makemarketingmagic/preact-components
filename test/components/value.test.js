import { h } from 'preact';
import render from 'preact-render-to-string'
import Value from './../../src/components/Value';

describe('Value', () => {
    it('should show the value passed to it', () => {
        const value = render(<Value>Hello World</Value>)
        expect(value).toMatchSnapshot()
    });
})