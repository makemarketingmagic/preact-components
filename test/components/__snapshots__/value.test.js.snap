// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Value should show the value passed to it 1`] = `<div class="value">Hello World</div>`;
